src/client/options/mod.rs

```diff
@@ -1890,6 +1898,9 @@ impl ConnectionString {
                         Some(index) => {
                             let (k, v) = exclusive_split_at(kvp, index);
                             let key = k.ok_or_else(err_func)?;
+                            if key == "ALLOWED_HOSTS" {
+                                return Err(err_func());
```

Comment:
"improperly formatted" might be confusing here -- can we add a message indicating that this val has to be specified through client options (if I'm understanding the spec correctly)?